### PR TITLE
feat(api): add endpoint to retrieve consensus contract addresses

### LINKED ISSUE #714: feat(api): add endpoint to retrieve consensus contract addresses
Add method and endpoint to return all the consensus contract addresses from the deployed contracts. This will allow clients to query the addresses of all consensus-related contracts through a simple API call.

### Tasks
1. Add method in ConsensusService to get all contract addresses
2. Create new GET endpoints to get consensus address

### Acceptance Criteria
- Endpoint successfully works
- Handles missing contracts gracefully
- Returns appropriate error messages when contracts aren't available
- Includes basic documentation for the endpoint

### Additional Notes
- This will be useful for the frontal application that need to interact with the consensus contracts
- Consider adding basic validation of addresses
- This issue depends on #713

### PR DESCRIPTION
**Pull request by mpaya5**
Fixes #714

# What

- Uses method `_load_contract` from ConsensusService to retrieve the contract address
- Created new RPC endpoint `eth_getContract` to expose contract aaddress
- Added error handling for missing contracts

# Why

- Frontend applications need to interact with consensus contracts and require their addresses
- Provides a new endpoint to access contract address and ABI address through the API
- Simplifies contract interaction by centralizing address management

# Testing done

- Tested endpoint with existing contracts (GhostContract, ConsensusManager, etc.)
- Verified error handling when requesting non-existent contracts
- Tested response format including contract name and address
- Verified endpoint accessibility through JSON-RPC at port 4000

# Decisions made

- Used `eth_getContract` as the endpoint name to maintain consistency with Ethereum naming conventions
- Return both contract address and ABI to provide complete contract information
- Implemented error handling using JSONRPCError for consistent error responses
- Kept the implementation simple by leveraging existing `_load_contract` method

# Checks

- [x] I have tested this code
- [x] I have reviewed my own PR
- [x] I have created an issue for this PR
- [x] I have set a descriptive PR title compliant with conventional commits

# Reviewing tips

- Key files to review:
  - `backend/protocol_rpc/endpoints.py` - New endpoint implementation
  - `backend/rollup/consensus_service.py` - Contract loading logic
- Test the endpoint with both valid and invalid contract names
- Verify error messages are helpful for debugging

# Depends on

That PR depends on #713 

# User facing release notes

Added new `eth_getContract` endpoint that allows retrieving deployed contract addresses and ABIs through the API. This enables frontend applications to easily access consensus contract information without hardcoding addresses. Simply call the endpoint with a contract name to get its deployment details.

### CODE REVIEW

**Comment by mpaya5** [comment]
Example usage:
```
curl -X POST http://localhost:4000/api \
-H "Content-Type: application/json" \
-d '{
  "jsonrpc": "2.0",
  "method": "eth_getContract",
  "params": ["GhostContract"],
  "id": 1
}'
```

You will receive:
```
{"id":{id},"jsonrpc":"2.0","result":{"address":"0xCf7Ed3AccA5a467e9e704C703E8D87F634fB0Fc9","contract_name":"GhostContract"}}
```

File: backend/protocol_rpc/endpoints.py [hunk 1]
```diff
@@ -655,3 +683,7 @@ def register_all_rpc_endpoints(
         partial(set_finality_window_time, consensus),
         method_name="sim_setFinalityWindowTime",
     )
+    register_rpc_endpoint(
+        partial(get_contract, consensus_service),
+        method_name="eth_getContract",
```
Comment (cristiam86) [inline backend/protocol_rpc/endpoints.py:688]: @mpaya5 this is not an Ethereum endpoint nor a GenLayer one. It is only a simulator's endpoint, so we should name it `sim_getContract`

**Review by cristiam86 — CHANGES_REQUESTED** [review]
Great, we just need to change the endpoint's name

File: backend/protocol_rpc/endpoints.py (hunk 1 above)
Comment (mpaya5) [reply backend/protocol_rpc/endpoints.py]: Done!

**Pull request merged**